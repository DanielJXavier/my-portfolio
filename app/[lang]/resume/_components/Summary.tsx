import { useParams } from "next/navigation";

import { getDictionary } from "get-dictionary";
import { Lang } from "i18n-config";

import { maxSummaryParagraphs } from "../_config";
import { useRef } from "react";

export default function Summary() {
  const { lang } = useParams<{ lang: Lang }>();

  const {
    home: { summary: defaultSummary },
    resume: { sectionTitles },
  } = getDictionary(lang);

  const summary = useRef(
    defaultSummary.filter((_, i) => i < maxSummaryParagraphs)
  );

  return (
    <section>
      <h2 className="text-[16pt] font-bold">{sectionTitles.summary}</h2>
      <textarea
        className="w-full text-[12pt] text-justify align-top border border-primary print:border-none print:resize-none"
        defaultValue={summary.current.join("\n")}
        rows={summary.current.reduce(
          (rows, paragraph) => rows + Math.ceil(paragraph.length / 8 / 12),
          0
        )}
      />
    </section>
  );
}
