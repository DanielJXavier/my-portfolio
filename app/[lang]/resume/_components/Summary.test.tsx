import "@testing-library/jest-dom";

import { render } from "@testing-library/react";

import Summary from "./Summary";

import { useParams } from "next/navigation";

jest.mock("next/navigation");

(useParams as jest.Mock).mockImplementation(() => ({ lang: "en" }));

describe("Summary component", () => {
  it("renders the component", () => {
    const { container } = render(<Summary />);

    expect(container).toMatchSnapshot();
  });
});
