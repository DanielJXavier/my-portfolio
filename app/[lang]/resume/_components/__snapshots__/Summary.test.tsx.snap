// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Summary component renders the component 1`] = `
<div>
  <section>
    <h2
      class="text-[16pt] font-bold"
    >
      Summary
    </h2>
    <textarea
      class="w-full text-[12pt] text-justify align-top border border-primary print:border-none print:resize-none"
      rows="11"
    >
      A Senior/Lead Front-End Engineer with 13+ years of experience helping US-based companies build intuitive user interfaces, ensuring code quality, high performance, scalability, accessibility, and security. Specialized in React, Next.js, TypeScript, and Tailwind CSS, now powered by top AI tools to go even faster.

Currently working at Lovevery, delivering meaningful projects like the New Checkout with a net gain of 7.3 new checkouts per day and the ERC launch to reach more than 455k customers in just a few months.

What I like most about Front-End is combining clean code, best practices, and design to create products that truly make a difference for real users and bring remarkable business results!
    </textarea>
  </section>
</div>
`;
